### ISSUE TITLE
`FileSystem.Exists(...)` returns wrong result on Windows

### ISSUE THREAD

**Issue by Rinkaa** [coding-jackalope/Slab#22]
This is tested with the `main.lua` in this repo. At the "Dialog" section of the test window, click "Open File" opening a file selection window, then try to open any file:
![slab](https://user-images.githubusercontent.com/8685221/83968319-f8f02700-a8fa-11ea-9f05-683986dbc40b.gif)
Any file is not able to be opened via either double-click or the "OK" button.

I made some test prints on local:
```lua
-- FileSystem.lua:219
function FileSystem.Exists(Path)
	print(Path) --debug
	local Handle = io.open(Path)
	if Handle ~= nil then
		print(Handle~=nil) --debug
		io.close(Handle)
		return true
	else
```
No output when I used double-click (perhaps it is not a bug, just me feeling a bit confused double-clicking can open folders but not files), and
here is the output when I used the "OK" button:

> z:/ParentNameParentNameParentNameParentName/TestFolder/Folder1/File1.txt/
> true
> z:/ParentNameParentNameParentNameParentName/TestFolder/Folder1/File2/
> true
> z:/ParentNameParentNameParentNameParentName/TestFolder/Folder2/File3.extrasuffix.txt/
> true
> z:/ParentNameParentNameParentNameParentName/TestFolder/File4/
> true

Since `FileSystem.IsDirectory(...)` directly depends on `FileSystem.Exists(...)`, I suppose this should be the reason.

This happens on my Windows 10. Then I copied all the code to my Android phone, and it seemed that files can be opened as expected there.

**Comment by coding-jackalope** [comment]
Hello, thanks for reporting this issue. I was able to reproduce the issue with double-click not selecting the file and closing the file dialog. However, the OK button selection seems to work fine for me. The double-click issue is related to the File Dialog logic only acting upon list item double-clicks for dialog entries and not files. This can be solved through properly handling list item double-clicks for files.

As for the FileSystem.Exists and FileSystem.IsDirectory functions, these should be rewritten to make use of the FFI system and properly call the OS-specific functions.

I am currently working on the next update. This issue along with the rewritten Exists/IsDirectory functions will be included in the update. I will also merge in some of the changes made in the pull request that fixes some issues and improves the file dialog behavior.

**Comment by coding-jackalope** [comment]
This issue has been resolved in the 0.7.0 release.
